src/client/containers/CartPage/CartPage.Container.js

```diff
@@ -66,54 +93,89 @@ const CartPageContainer = ({ isAuthenticated }) => {
     const productId = cartItem[index].productId;
     changeProductQuantity(productId, 0);
   };
+
+  const checkFavortie = (productId) => {
+    if (favorite.length > 0) {
+      return (
+        favorite.filter((item) => item.id === parseInt(productId)).length > 0
+      );
+    }
+    return false;
+  };
+
+  const showCartItems = () => {
+    if (cartItem.length === 0) {
+      return <h1> Your Cart is Empty </h1>;
+    }
+
+    return cartItem.map((item, index) => {
+      return (
+        <ShoppingItem
+          productId={item.productId}
+          productName={item.name}
+          quantity={item.stock_amount}
+          price={item.price}
+          productImg={item.picture}
+          initValue={item.quantity}
+          getQuantity={(value) => {
+            getItemQuantity(index, value);
+          }}
+          onDelete={() => handleOnDeleteItem(index)}
+          userId={userId}
+          isFavorite={checkFavortie(item.productId)}
+        />
+      );
+    });
+  };
+  const showDeliveryInfo = () => {
+    if (userId) {
+      return <DeliveryInfoV2 editMode={true} vertDisplay={false} user={user} />;
+    }
+  };
+  const showContactDetails = () => {
+    if (userId) {
+      return (
+        <ContactForm
+          fullName={auth.currentUser.displayName}
+          email={auth.currentUser.email}
+        />
+      );
+    }
+  };
+
   return (
-    <div>
-      <h1>SHOPPING CART</h1>
-      <div className="main">
-        <div className="left">
-          <div className="item-list">
-            {IsLoading ? (
-              <Loader />
-            ) : (
-              cartItem.map((item, index) => {
-                return (
-                  <ShoppingItem
```

Comment:
It is good also to add a **key** to this map, so it does not show warning in console.